s3fs/core.py

```diff
@@ -450,6 +453,37 @@ def ls(self, path, detail=False, refresh=False, **kwargs):
         else:
             return [f['Key'] for f in files]
 
+    def isdir(self, path, refresh=False):
+        """ Check if path points to a directory.
+
+        Check is cached unless `refresh=True`.
+
+        Parameters
+        ----------
+        path : string/bytes
+            location to check
+        refresh : bool
+            If true, don't look in the info cache
+        """
+        if path.startswith('s3://'):
+            path = path[len('s3://'):]
+        path = path.rstrip('/')
+        return not path or bool(self._lsdir(path, refresh=refresh, max_items=1))
```

Comment:
So the logic is, if it contains anything, it must be a directory. I think it would be equally had from the listing of the parent directory: it should turn up in the CommonPrefixes and be marked as a directory.

Reply:
If I understand what you mean, this would require listing the whole parent directory (no `max_items`) which is potentially more expensive.

Reply:
Yes, it would mean listing the parent directory. There is a good chance that the parent directory is cached, though, so it could involve no download at all. Perhaps we could check for the cached listing and use it if it exists, or use this method if it does not?

Reply:
Good idea, implemented at ceb7def3d9e897c0fe3d4774b87ed61a70dadb2b.